cs4660/graph/graph.py

```diff
@@ -113,26 +180,83 @@ def __init__(self):
         self.nodes = []
 
     def adjacent(self, node_1, node_2):
-        pass
+        if node_1 not in self.nodes or node_2 not in self.nodes:
+            return False
+        index1 = self.__get_node_index(node_1)
+        index2 = self.__get_node_index(node_2)
+        if self.adjacency_matrix[index1][index2] is not None or self.adjacency_matrix[index2][index1] is not None:
```

Comment:
adjacent should check the direction, in other word, the second check here is not necessary.